File: Java_Programs/Calculator.java
import java.util.Scanner;

public class Calculator {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter expression (e.g. 2.5 + 3): ");
        double a, b; String op;
        if (!sc.hasNextDouble()) { System.err.println("Invalid input"); sc.close(); System.exit(1); }
        a = sc.nextDouble();
        if (!sc.hasNext()) { System.err.println("Invalid input"); sc.close(); System.exit(1); }
        op = sc.next();
        if (!sc.hasNextDouble()) { System.err.println("Invalid input"); sc.close(); System.exit(1); }
        b = sc.nextDouble();
        sc.close();
        double r = 0; boolean ok = true;
        switch (op) {
            case "+": r = a + b; break;
            case "-": r = a - b; break;
            case "*": r = a * b; break;
            case "/": if (b == 0) { System.err.println("Division by zero"); System.exit(1); } r = a / b; break;
            default: ok = false;
        }
        if (!ok) { System.err.println("Unsupported operator: " + op); System.exit(1); }
        System.out.println("Result: " + r);
    }
}

File: C_Programs/calculator.c
#include <stdio.h>

int main(void) {
    double a, b;
    char op;
    printf("Enter expression (e.g. 2.5 + 3): ");
    if (scanf("%lf %c %lf", &a, &op, &b) != 3) {
        fprintf(stderr, "Invalid input\n");
        return 1;
    }
    double r;
    int ok = 1;
    switch (op) {
        case '+': r = a + b; break;
        case '-': r = a - b; break;
        case '*': r = a * b; break;
        case '/': if (b == 0) { fprintf(stderr, "Division by zero\n"); return 1; } r = a / b; break;
        default: ok = 0;
    }
    if (!ok) { fprintf(stderr, "Unsupported operator: %c\n", op); return 1; }
    printf("Result: %g\n", r);
    return 0;
}
